lib/sinatra/main.rb

```diff
@@ -29,7 +34,11 @@ class Application < Base
 
     set :run, Proc.new { File.expand_path($0) == File.expand_path(app_file) }
 
-    ParamsConfig.each { |k, v| set k, v } if run? && ARGV.any?
+    if run? && ARGV.any?
+      error = ParamsConfig.delete(:optparse_error)
+      raise error if error
```

Comment:
Do we need to raise an invalid option error, or could we just swallow and ignore?

Reply:
We should raise this.
Because if we ignore that, actual invalid error can not be handled.
Developer may use "require "sinatra" with "foo.rb" then exec "ruby foo.rb --opts"

If the command are "ruby foo.rb --invalidopts", then sinatra should raise an error.